src/ucp/proto/proto_debug.c

```diff
@@ -190,6 +192,76 @@ static int ucp_proto_debug_is_info_enabled(ucp_context_h context,
     return fnmatch(proto_info_config, select_param_str, FNM_CASEFOLD) == 0;
 }
 
+static int
+ucp_proto_debug_get_range_info(ucp_worker_h worker,
+                               const ucp_proto_select_elem_t *select_elem,
+                               size_t msg_length, size_t *range_end,
+                               ucp_proto_query_attr_t *proto_attr)
+{
+    const ucp_proto_perf_range_t *range;
+
+    range      = ucp_proto_perf_range_search(select_elem, msg_length);
+    *range_end = range->max_length;
```

Comment:
can we say this fix is mostly this (set range_end to range->max_length instead of proto_attr.max_msg_length) and the rest is mostly refactoring?
In this case, can we fix rndv protocol query to set proto_attr.max_msg_length in a way that will distinguish between different remote protocols?

Reply:
If we only change `proto_attr.max_msg_length` by `range->max_length`, then we will get many duplicated rows. Because there can be many ranges for the same combination of protocol and device.

Yes, I agree. The query for rendezvous should be fixed instead. Currently we set `ucp_proto_query_attr_t::max_msg_length` to `SIZE_MAX` in query method. Which is not correct.

Reply:
yes, we need to set it to the max range that the same protocol (including remote) is used. this way, will also not have duplicates.

Reply:
Fixed query.